### PR TITLE
*: add safe mode test

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 
ref: https://internal.pingcap.net/jira/browse/TOOL-790

### What is changed and how it works?
we use gofail to simulate DM-worker exits in two scene:
* DM-worker exits when it is in sharding re-sync after sharding group synced
* DM-worker exits when waiting for sharding group synced

### Check List 

Tests 

 - Integration test

### CODE REVIEW

File: tests/safe_mode/dmctl.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply tests/safe_mode/dmctl.go]: we have extract some dmctl API in `tests/utils/dmctl.go`, this `dmctl.go` is simply to call encapsulated function and add different control flow

File: tests/safe_mode/run.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+DB1_PORT=3306
+DB2_PORT=3307
+TIDB_PORT=4000
+MASTER_PORT=8261
+WORKER1_PORT=8262
+WORKER2_PORT=8263
+WORK_DIR=$TEST_DIR/safe_mode
+
+# we do clean staff at beginning of each run, so we can keep logs of the latset run
+function cleanup1() {
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply tests/safe_mode/run.sh]: These cleanup functions are not commonly feature, we may refine it later

File: tests/safe_mode/data/db1.increment.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
+use safe_mode_test;
+insert into t1 (uid, name) values (10003, 'Buenos Aires');
+update t1 set name = 'Gabriel José de la Concordia García Márquez' where `uid` = 10001;
+update t1 set name = 'One Hundred Years of Solitude' where name = 'Cien años de soledad';
+alter table t1 add column age int;
+alter table t2 add column age int;
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply tests/safe_mode/data/db1.increment.sql]: added

**Comment by csuzhangxc** [comment]
LGTM

**Pull request closed without merging**